Lib/zipfile.py

```diff
@@ -484,6 +485,13 @@ def _decodeExtra(self):
                 except struct.error:
                     raise BadZipFile(f"Corrupt zip64 extra field. "
                                      f"{field} not found.") from None
+            elif tp == 0x7075:
+                data = extra[4:ln+4]
+                # Unicode Path Extra Field
+                up_version, up_name_crc = unpack('<BL', data[:5])
```

Comment:
As we're accepting user entered data here we should check the data isn't corrupt. Below is my (untested) suggested approach. It adds warnings for a corrupt field rather than raising an exception to keep behaviour the same with previous versions of Python which ignored these fields.

```
try:
    up_version, up_name_crc = unpack('<BL', data[:5])
    if up_version == 1 and up_name_crc == self.orig_filename_crc:
        up_unicode_name = data[5:].decode('utf-8')
        if up_unicode_name:
            self.filename = up_unicode_name
        else:
            import warnings
            warnings.warn('Empty UTF path extra field (0x7075)', stacklevel=2)
except struct.error:
    import warnings
    warnings.warn('Corrupt UTF path extra field (0x7075)', stacklevel=2)
except UnicodeDecodeError:
    import warnings
    warnings.warn('Corrupt UTF path extra field (0x7075) - invalid UTF bytes', stacklevel=2)
```